theano/sandbox/gpuarray/opt.py

```diff
@@ -751,6 +751,14 @@ def local_gpua_dot22(node, context_name):
 
 
 @register_opt('fast_compile')
+@op_lifter([tensor.blas.Dot22Scalar])
+def local_gpua_dot22scalar(node, context_name):
+    dot = gpu_dot22(as_gpuarray_variable(node.inputs[0], context_name),
+                    as_gpuarray_variable(node.inputs[1], context_name))
+    return [node.inputs[2] * dot]
```

Comment:
why not convert this to a gemm call with alloc_empty?

Add a test to make sure it don't get broken.